File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Reflection/GetDeclaration/TypeInfo.GetShortDeclaraction.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_813

    <System.Runtime.CompilerServices.Extension> _
    Friend Function GetShortDeclaraction(this As Type) As String
        If this = GetType(Boolean) Then
            Return "bool"
        End If
        If this = GetType(Byte) Then
            Return "byte"
        End If
        If this = GetType(Char) Then
            Return "char"
        End If
        If this = GetType(Decimal) Then
            Return "decimal"
        End If
        If this = GetType(Double) Then
            Return "double"
        End If
        If this = GetType([Enum]) Then
            Return "enum"
        End If
        If this = GetType(Single) Then
            Return "float"
        End If
        If this = GetType(Integer) Then
            Return "int"
        End If
        If this = GetType(Long) Then
            Return "long"
        End If
        If this = GetType(Object) Then
            Return "object"
        End If
        If this = GetType(SByte) Then
            Return "sbyte"
        End If
        If this = GetType(Short) Then
            Return "short"
        End If
        If this = GetType(String) Then
            Return "string"
        End If
        If this = GetType(UInteger) Then
            Return "uint"
        End If
        If this = GetType(ULong) Then
            Return "ulong"
        End If
        If this = GetType(UShort) Then
            Return "ushort"
        End If

        Return this.Name
    End Function
End Module

File: src/Z.Reflection/GetDeclaration/_Helper/TypeInfo.GetShortDeclaraction.cs
// Copyright (c) 2015 ZZZ Projects. All rights reserved
// Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
// Website: http://www.zzzprojects.com/
// Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
// All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

using System;

public static partial class Extensions
{
    /// <summary>A Type extension method that gets short declaraction.</summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>The short declaraction.</returns>
    internal static string GetShortDeclaraction(this Type @this)
    {
        if (@this == typeof (bool))
        {
            return "bool";
        }
        if (@this == typeof (byte))
        {
            return "byte";
        }
        if (@this == typeof (char))
        {
            return "char";
        }
        if (@this == typeof (decimal))
        {
            return "decimal";
        }
        if (@this == typeof (double))
        {
            return "double";
        }
        if (@this == typeof (Enum))
        {
            return "enum";
        }
        if (@this == typeof (float))
        {
            return "float";
        }
        if (@this == typeof (int))
        {
            return "int";
        }
        if (@this == typeof (long))
        {
            return "long";
        }
        if (@this == typeof (object))
        {
            return "object";
        }
        if (@this == typeof (sbyte))
        {
            return "sbyte";
        }
        if (@this == typeof (short))
        {
            return "short";
        }
        if (@this == typeof (string))
        {
            return "string";
        }
        if (@this == typeof (uint))
        {
            return "uint";
        }
        if (@this == typeof (ulong))
        {
            return "ulong";
        }
        if (@this == typeof (ushort))
        {
            return "ushort";
        }

        return @this.Name;
    }
}